src/common/searchMovies.ts

```diff
@@ -0,0 +1,70 @@
+import { DetailedMovie } from "@/common/types/models";
+
+/**
+ * 
+ * @param reviews 
+ * @param searchQuery 
+ * @returns reviews filtered by searchQuery.
+ * 
+ * You can apply filters on the searchQuery with text:value. For example, to filter by title and genre, you can use:
+ * 
+ * "title:jaws genre:horror"
+ * 
+ * Incluidng multiple filters seperated by spaces will implicitly do an AND search between them.
+ * 
+ * TODO: Add support for OR searches.
+ * TODO: Create a new vue component for the search bar that highlights filters different colors.
+ * TODO: Make the watchlist and backlog use DetailedMovie[] so they can use the same search function and bar.
+ * 
+ */
+export function filterMovies(reviews: DetailedMovie[], searchQuery: string): DetailedMovie[] {
+    if (!searchQuery) {
```

Comment:
Just wondering if this check is needed? By the type definition, it should never be undefined. If we're allowing it to be undefined, we should probably change the type definition to reflect that